walkthroughs/week-3-server/examples/random-quotes/src/main/webapp/script.js

```diff
@@ -49,15 +49,33 @@ function addQuoteToDom(quote) {
   quoteContainer.innerText = quote;
 }
 
+function getRandomQuoteUsingPromiseChain() {
+  fetch('/random-quote')
+  .then(function(response) { 
+    return response.text() 
+  })
+  .then(function(quote) {
+    document.getElementById('quote-container').innerText = quote;
+  })
+  .catch(function(err) {
+    console.log("Error: " + err);
+  });
+}
+
 /**
  * The above code is organized to show each individual step, but we can use an
  * ES6 feature called arrow functions to shorten the code. This function
  * combines all of the above code into a single Promise chain. You can use
  * whichever syntax makes the most sense to you.
  */
 function getRandomQuoteUsingArrowFunctions() {
-  fetch('/random-quote').then(response => response.text()).then((quote) => {
+  fetch('/random-quote')
+  .then(response => response.text())
+  .then(quote => {
     document.getElementById('quote-container').innerText = quote;
+  })
+  .catch(err => {
+    console.log("Error: " + err);
```

Comment:
what will users see in this case? an error page? Or do you need a backup value for the innerText?

Reply:
Very good point. I added a backup value for the innerText. For the future in a real implementation, I think an error page would be best. Please let me know if you have a specific preference/suggestion.